src/core/MOM_barotropic.F90

```diff
@@ -4960,30 +5127,96 @@ subroutine barotropic_init(u, v, h, eta, Time, G, GV, US, param_file, diag, CS,
         call MOM_read_data(wave_drag_file, wave_drag_u, CS%lin_drag_u, G%Domain, &
-                           position=EAST_FACE, scale=GV%m_to_H*US%T_to_s)
+                           position=EAST_FACE, scale=wave_drag_scale*GV%m_to_H*US%T_to_s)
         call pass_var(CS%lin_drag_u, G%Domain)
-        CS%lin_drag_u(:,:) = wave_drag_scale * CS%lin_drag_u(:,:)
 
         call MOM_read_data(wave_drag_file, wave_drag_v, CS%lin_drag_v, G%Domain, &
-                           position=NORTH_FACE, scale=GV%m_to_H*US%T_to_s)
+                           position=NORTH_FACE, scale=wave_drag_scale*GV%m_to_H*US%T_to_s)
         call pass_var(CS%lin_drag_v, G%Domain)
-        CS%lin_drag_v(:,:) = wave_drag_scale * CS%lin_drag_v(:,:)
 
-      else
+      elseif (len_trim(wave_drag_var) > 0) then
         allocate(lin_drag_h(isd:ied,jsd:jed), source=0.0)
 
-        call MOM_read_data(wave_drag_file, wave_drag_var, lin_drag_h, G%Domain, scale=GV%m_to_H*US%T_to_s)
+        call MOM_read_data(wave_drag_file, wave_drag_var, lin_drag_h, G%Domain, &
+                           scale=wave_drag_scale*GV%m_to_H*US%T_to_s)
         call pass_var(lin_drag_h, G%Domain)
         do j=js,je ; do I=is-1,ie
-          CS%lin_drag_u(I,j) = wave_drag_scale * 0.5 * (lin_drag_h(i,j) + lin_drag_h(i+1,j))
+          CS%lin_drag_u(I,j) = 0.5 * (lin_drag_h(i,j) + lin_drag_h(i+1,j))
         enddo ; enddo
         do J=js-1,je ; do i=is,ie
-          CS%lin_drag_v(i,J) = wave_drag_scale * 0.5 * (lin_drag_h(i,j) + lin_drag_h(i,j+1))
+          CS%lin_drag_v(i,J) = 0.5 * (lin_drag_h(i,j) + lin_drag_h(i,j+1))
         enddo ; enddo
         deallocate(lin_drag_h)
-      endif ! len_trim(wave_drag_u) > 0 .and. len_trim(wave_drag_v) > 0
-    endif ! len_trim(wave_drag_file) > 0
-  endif ! CS%linear_wave_drag
+      endif ! (len_trim(wave_drag_u) > 0 .and. len_trim(wave_drag_v) > 0)
+    endif ! (len_trim(wave_drag_file) > 0)
+  endif ! (CS%linear_wave_drag)
+
+  if (CS%linear_freq_drag) then
+    ALLOC_(CS%lin_drag_um2(IsdB:IedB,jsd:jed)) ; CS%lin_drag_um2(:,:) = 0.0
+    ALLOC_(CS%lin_drag_vm2(isd:ied,JsdB:JedB)) ; CS%lin_drag_vm2(:,:) = 0.0
+    ALLOC_(CS%lin_drag_uk1(IsdB:IedB,jsd:jed)) ; CS%lin_drag_uk1(:,:) = 0.0
+    ALLOC_(CS%lin_drag_vk1(isd:ied,JsdB:JedB)) ; CS%lin_drag_vk1(:,:) = 0.0
+
+    if (len_trim(wave_drag_file) > 0) then
+      if (.not. CS%linear_wave_drag) then
+        inputdir = "." ;  call get_param(param_file, mdl, "INPUTDIR", inputdir)
+        wave_drag_file = trim(slasher(inputdir))//trim(wave_drag_file)
+        call log_param(param_file, mdl, "INPUTDIR/BT_WAVE_DRAG_FILE", wave_drag_file)
+      endif
+
+      if (CS%use_filter_m2 .and. m2_drag_scale > 0.0) then
+        if (len_trim(m2_drag_u) > 0 .and. len_trim(m2_drag_v) > 0) then
+          call MOM_read_data(wave_drag_file, m2_drag_u, CS%lin_drag_um2, G%Domain, &
+                             position=EAST_FACE, scale=m2_drag_scale*GV%m_to_H*US%T_to_s)
+          call pass_var(CS%lin_drag_um2, G%Domain)
```

Comment:
Doing a separate pass_var for `lin_drag_um2` and `lin_drag_vm2` will result in fields that do not rotate properly and will be incorrect at a norther tripolar fold.  Moreover, as these variables are at the u- and v-points, it is incorrect to use pass_var calls without a flag to indicate this, as that routine assumes by default that these variables are at the h-points.  The correct thing to do here would be to replace the two `pass_var()` calls at lines 5171 and 5175 with a single call to `pass_vector(m2_drag_u, m2_drag_v, G%domain, direction=To_All+SCALAR_PAIR)`.  The scalar pair here is because these are non-negative magnitudes and not components of a vector.

  Similar changes should also be made to the `pass_var()` calls on lines 5197 and 5201 in this PR, and also to the calls on lines 5131 and 5135 that were introduced with PR #703.  There is no reason to add a flag to recover the previous answers, because they could not pass the rotational symmetry testing and hence are demonstrably wrong.

Reply:
Thanks for the explanation. This has been corrected.